dojo/tools/noseyparker/parser.py

```diff
@@ -0,0 +1,101 @@
+                    # The following path is to account for the variability in the JSON lines output
+                    num_elements = len(match['provenance']) - 1
+                    json_path = match['provenance'][num_elements]
+
+                    title = f"Secret(s) Found in Repository with Commit ID {json_path['commit_provenance']['commit_metadata']['commit_id']}"
+                    filepath = json_path['commit_provenance']['blob_path']
+                    line_num = match['location']['source_span']['start']['line']
+                    description = f"Secret found of type:   {rule_name} \n" \
+                                  f"SECRET starts with:  '{secret[:3]}' \n" \
+                                  f"Committer Name: {json_path['commit_provenance']['commit_metadata']['committer_name']}  \n" \
+                                  f"Committer Email: {json_path['commit_provenance']['commit_metadata']['committer_email']} \n" \
+                                  f"Commit ID: {json_path['commit_provenance']['commit_metadata']['commit_id']}  \n" \
+                                  f"Location: {filepath} line #{line_num} \n " \
+                                  f"Line #{line_num} \n " \
+                                  f"Code Snippet Containing Secret: {match['snippet']['before']}***SECRET***{match['snippet']['after']} \n"
+
+                    # Internal de-duplication
+                    key = hashlib.md5((filepath + "|" + secret + "|" + str(line_num)).encode("utf-8")).hexdigest()
+
+                    # If secret already exists with the same filepath/secret/linenum
+                    if key in dupes:
+                        finding = dupes[key]
+                        finding.nb_occurences += 1
+                        dupes[key] = finding
+                    else:
+                        dupes[key] = True
+                        # Create Finding object
+                        finding = Finding(
+                            test=test,
+                            cwe=798,
+                            title=title,
+                            description=description,
+                            severity='High',
+                            mitigation="Reset the account/token and remove from source code. Store secrets/tokens/passwords in secret managers or secure vaults.",
+                            date=datetime.today().strftime("%Y-%m-%d"),
+                            verified=False,
+                            active=True,
+                            is_mitigated=False,
+                            file_path=filepath,
+                            line=line_num,
+                            static_finding=True,
+                            nb_occurences=1,
+                            dynamic_finding=False
+
+                        )
+                        dupes[key] = finding
+        else:
+            raise ValueError("JSON lines format not recognized. Make sure to use Nosey Parker v0.16.0")
```

Comment:
Probably good to mention explicitly that we're looking for a `.jsonl` file here.

```suggestion
            raise ValueError("JSON lines format not recognized (.jsonl file extension). Make sure to use Nosey Parker v0.16.0")
```